index.js

```diff
@@ -67,7 +96,30 @@ function getAverageIMDBRating() {}
  *  countByRating(movies);
  *  //> { G: 3, PG: 7 }
  */
-function countByRating() {}
+function countByRating(movies) {
+  let movieKeys = {};
+  if (movies.length === 0) {
+    return movieKeys;
+  }
+  let count = 0;
+  let movieRated = [];
+  for (let i = 0; i < movies.length; i++){
+    movieRated.push(movies[i].rated);
+  }
+  movieRated.sort();
+  let firstRated = movieRated[0];
+  for (let i = 0; i < movieRated.length; i++){
+    if (movieRated[i] === firstRated){
+      count ++;  
+    } else {
+      count = 1;
+      firstRated = movieRated[i];
+    }
+    movieKeys[movieRated[i]] = count;
+  }
```

Comment:
Nice job with the logic - especially the `.sort()` call.  

However there is a pattern here that is much quicker to implement by checking if `movieKeys` has certain keys.
In one loop we can use bracket notation to check if our object has a key that matches our `movies[i]` object.

` if (movieKeys[movies[i].rated]) `  will check if the object has a key that matches our current `movie[i].rated`.  See if you can use that solution to refactor this with only one loop.  This is a very good pattern to get used to in writing efficient
 code.